packages/testing/containers/services/types.ts

```diff
@@ -0,0 +1,75 @@
+import type { StartedTestContainer, StartedNetwork } from 'testcontainers';
+
+export interface FileToMount {
+	content: string;
+	target: string;
+}
+
+export interface ServiceMeta {
+	/**
+	 * Files to mount into n8n containers. Use when n8n needs files that can't
+	 * be passed via environment (e.g., NODE_EXTRA_CA_CERTS requires a file path).
+	 * See keycloak.ts for usage example.
+	 */
+	n8nFilesToMount?: FileToMount[];
+}
+
+export interface ServiceResult<TMeta = unknown> {
+	container: StartedTestContainer;
+	meta: TMeta;
+}
+
+export interface StartContext {
+	config: StackConfig;
+	projectName: string;
+	mains: number;
+	workers: number;
+	isQueueMode: boolean;
+	usePostgres: boolean;
+	needsLoadBalancer: boolean;
+	environment: Record<string, string>;
+	serviceResults: Record<string, ServiceResult>;
+	allocatedPorts: { main?: number; loadBalancer?: number };
+	baseUrl?: string;
+}
+
+export interface StackConfig {
+	mains?: number;
+	workers?: number;
+	postgres?: boolean;
+	env?: Record<string, string>;
+	projectName?: string;
+	resourceQuota?: { memory?: number; cpu?: number };
+	services?: string[];
+}
+
+export interface Service<TResult extends ServiceResult = ServiceResult> {
```

Comment:
Can we have a JSDoc for each of these interface methods to describe what they should be used for?